pkg/flare/archive.go

```diff
@@ -61,7 +62,7 @@ func CompleteFlare(fb flaretypes.FlareBuilder, diagnoseDeps diagnose.SuitesDeps)
 		fb.AddFileFromFunc("tagger-list.json", getAgentTaggerList)                      //nolint:errcheck
 		fb.AddFileFromFunc("workload-list.log", getAgentWorkloadList)                   //nolint:errcheck
 		fb.AddFileFromFunc("process-agent_tagger-list.json", getProcessAgentTaggerList) //nolint:errcheck
-		if !config.Datadog().GetBool("process_config.run_in_core_agent.enabled") {
+		if !processutil.GetRunInCoreAgentConfig(config.Datadog()) {
```

Comment:
The issue here is that you're making the flare comp depends on the process/util folder which brings docker, workloadmeta, agent-payloads and way more as dependencies for the flare.

`CompleteFlare` is a legacy function aimed at finishing the flare until our code base fully used component. The best woiuld be to migrate `getChecksFromProcessAgent` to a [flare provider](https://datadoghq.dev/datadog-agent/components/shared_features/flares/) somewhere in comp/process. If we don't want to do this, we have the process comp set `run_in_core_agent.enabled` to false in the configuration when starting (that way we don't have to use `GetRunInCoreAgentConfig` at all.

Reply:
I see. I think I'll avoid creating a flareprovider for that as we want to move away from running the checks on the process agent regardless. Setting the config within the component seems like a better approach for handling this.

Reply:
> The issue here is that you're making the flare comp depends on the process/util folder which brings docker, workloadmeta, agent-payloads and way more as dependencies for the flare.

@hush-hush Is there some way to see these changes to the dependency tree for a PR? I had discussed this with Daniel, and previously there was a message from `pr-commenter` that showed no major changes to package imports, so we were misled to think that there wasn't any impact.

Reply:
@robertjli I don't know, that's a good question but it would be for devX.